### PR TITLE
feat: add browser_type.connect

### PR DESCRIPTION
**Pull request by kumaraditya303**

### CODE REVIEW

**Review by mxschmitt — CHANGES_REQUESTED** [review]
- unit tests are missing (a lot of edge cases) see python for how to create the test infra
- close event handling is missing (browser, context,page) gets close event when ws commection closes
- request headers are missing
- slowmo is missing
- https://github.com/microsoft/playwright/blob/a8d48a1a48e2a31638f61a467455fb322cb7228b/src/client/browser.ts#L101 is missing

File: transport.go [hunk 1]
```diff
@@ -9,17 +9,63 @@ import (
...
+
+func (t *websocketTransport) Start() error {
+	conn, _, err := websocket.DefaultDialer.Dial(t.url, nil)
+	if err != nil {
+		return fmt.Errorf("could not connect to websocket: %w", err)
+	}
+	t.conn = conn
+	for {
+		msg := &message{}
+		err := t.conn.ReadJSON(msg)
```
Comment (mxschmitt) [inline transport.go:44]: Single error returns get usually handled inside the if condition. It's shorter. 

```suggestion
		if err := t.conn.ReadJSON(msg); err != nil {
       ...
```

Comment (kumaraditya303) [reply transport.go:44]: Next time

File: transport.go [hunk 2]
```diff
@@ -9,17 +9,66 @@ import (
...
+	}
+	return nil
+}
+
+func (t *webSocketTransport) SetDispatch(dispatch func(msg *message)) {
+	t.dispatch = dispatch
+}
+
+func (t *webSocketTransport) Stop() error {
+	_ = t.conn.Close()
```
Comment (kumaraditya303) [inline transport.go:66]: This is a problem because cannot get how the connection was closed ?

**Comment by kumaraditya303** [comment]
Windows is fixed now but linux is still not working

File: tests/remote_server_linux_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	}
+}
+
+func (s *remoteServer) Close() {
+	cmd := exec.Command("kill", "-KILL", strconv.Itoa(-s.cmd.Process.Pid))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	_ = cmd.Run()
+	_ = s.cmd.Wait()
+	time.Sleep(time.Second)
```
Comment (kumaraditya303) [inline tests/remote_server_linux_test.go:54]: Delay required to kill the browser

**Comment by kumaraditya303** [comment]
> * unit tests are missing (a lot of edge cases) see python for how to create the test infra
> * close event handling is missing (browser, context,page) gets close event when ws commection closes
> * request headers are missing
> * slowmo is missing
> * https://github.com/microsoft/playwright/blob/a8d48a1a48e2a31638f61a467455fb322cb7228b/src/client/browser.ts#L101 is missing

The missing headers and slow can be implemented later on

File: tests/remote_server_windows_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	if err != nil {
+		log.Fatalf("could not read url: %v", err)
+	}
+	return &remoteServer{
+		url: url,
+		cmd: cmd,
+	}
+}
+
+func (s *remoteServer) Close() {
```
Comment (mxschmitt) [inline tests/remote_server_windows_test.go:47]: Ideally we have something like a "process launcher" which has this specific kill logic for linux/windows in place and not duplicate the whole remote server logic.
